test/support/mocks/graph-client.service.mock.ts

```diff
@@ -0,0 +1,57 @@
+import { Client, GraphRequest } from '@microsoft/microsoft-graph-client';
+import { when } from 'jest-when';
+
+class MockClient {
+  api: jest.Mock;
+  constructor() {
+    this.api = jest.fn();
+  }
+}
+
+class MockRequest {
+  get: jest.Mock<any, any, any>;
+  filter: jest.Mock<any, any, any>;
+  expand: jest.Mock<any, any, any>;
+  constructor() {
+    this.get = jest.fn();
+
+    this.filter = jest.fn();
+    this.expand = jest.fn();
+  }
+}
+
+export class MockGraphClientService {
+  client: Client;
+  request: GraphRequest;
+  constructor() {
+    this.client = new MockClient() as unknown as Client;
+    this.request = new MockRequest() as unknown as GraphRequest;
+  }
+
+  getApiRequestObject() {
+    return this.request;
+  }
+
+  mockSuccessfulGraphApiCallWithPath(path: string) {
+    when(this.client.api).calledWith(path).mockReturnValueOnce(this.request);
+    return this;
+  }
+
+  mockSuccessfulFilterCallWithFilterString(filterString: string) {
+    when(this.request.filter).calledWith(filterString).mockReturnValueOnce(this.request);
+    return this;
+  }
+
+  mockSuccessfulExpandCallWithExpandString(expandString: string) {
+    when(this.request.expand).calledWith(expandString).mockReturnValueOnce(this.request);
+    return this;
+  }
+
+  mockSuccessfulGraphGetCall(response) {
```

Comment:
Can we have a type for the input here please?

Reply:
I've changed this to be a generic method as this can return potentially anything from Microsoft Graph